build_tools/build_detect_platform

```diff
@@ -45,8 +45,13 @@ if test -z "$OUTPUT"; then
   exit 1
 fi
 
-# we depend on C++11
-PLATFORM_CXXFLAGS="-std=c++11"
+# we depend on C++11, but should be compatible with newer standards
+if [ "$ROCKSDB_CXX_STANDARD" ]; then
```

Comment:
Are there backwards-incompatible changes between C++11 and newer standards then?

Reply:
Hopefully my wording is not so strong as to suggest someone can compile RocksDB with an overridden standard and does not need to bother running the test suite before considering it usable in production.